File: core/src/main/java/com/sequenceiq/cloudbreak/converter/CloudbreakEventToJsonConverter.java
package com.sequenceiq.cloudbreak.converter;

import org.springframework.stereotype.Component;

import com.sequenceiq.cloudbreak.api.model.CloudbreakEventsJson;
import com.sequenceiq.cloudbreak.domain.CloudbreakEvent;

@Component
public class CloudbreakEventToJsonConverter extends AbstractConversionServiceAwareConverter<CloudbreakEvent, CloudbreakEventsJson> {

    @Override
    public CloudbreakEventsJson convert(CloudbreakEvent entity) {
        CloudbreakEventsJson json = new CloudbreakEventsJson();
        json.setAccount(entity.getAccount());
        json.setBlueprintId(entity.getBlueprintId());
        json.setBlueprintName(entity.getBlueprintName());
        json.setCloud(entity.getCloud());
        json.setEventMessage(entity.getEventMessage());
        json.setEventType(entity.getEventType());
        json.setEventTimestamp(entity.getEventTimestamp().getTime());
        json.setRegion(entity.getRegion());
        json.setOwner(entity.getOwner());
        json.setStackId(entity.getStackId());
        json.setStackName(entity.getStackName());
        json.setStackStatus(entity.getStackStatus());
        json.setNodeCount(entity.getNodeCount());
        json.setInstanceGroup(entity.getInstanceGroup());
        json.setClusterStatus(entity.getClusterStatus());
        json.setClusterId(entity.getClusterId());
        json.setClusterName(entity.getClusterName());
        return json;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/CloudbreakEventToJsonConverter.kt
package com.sequenceiq.cloudbreak.converter

import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.domain.CloudbreakEvent
import com.sequenceiq.cloudbreak.api.model.CloudbreakEventsJson

@Component
class CloudbreakEventToJsonConverter : AbstractConversionServiceAwareConverter<CloudbreakEvent, CloudbreakEventsJson>() {

    override fun convert(entity: CloudbreakEvent): CloudbreakEventsJson {
        val json = CloudbreakEventsJson()
        json.account = entity.account
        json.blueprintId = entity.blueprintId
        json.blueprintName = entity.blueprintName
        json.cloud = entity.cloud
        json.eventMessage = entity.eventMessage
        json.eventType = entity.eventType
        json.eventTimestamp = entity.eventTimestamp.time
        json.region = entity.region
        json.owner = entity.owner
        json.stackId = entity.stackId
        json.stackName = entity.stackName
        json.stackStatus = entity.stackStatus
        json.nodeCount = entity.nodeCount
        json.instanceGroup = entity.instanceGroup
        json.clusterStatus = entity.clusterStatus
        json.clusterId = entity.clusterId
        json.clusterName = entity.clusterName
        return json
    }
}
